presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/SimpleTTLNodeSelector.java

```diff
@@ -0,0 +1,291 @@
+        this.session = requireNonNull(session, "session is null");
+        checkArgument(session.getResourceEstimates().getExecutionTime().isPresent(), "Estimated execution time is not present");
+        estimatedExecutionTime = session.getResourceEstimates().getExecutionTime().get();
+        this.nodeMap = new AtomicReference<>(requireNonNull(nodeMap, "nodeMap is null"));
+        this.minCandidates = minCandidates;
+        this.includeCoordinator = includeCoordinator;
+        this.maxSplitsPerNode = maxSplitsPerNode;
+        this.maxPendingSplitsPerTask = maxPendingSplitsPerTask;
+        this.maxTasksPerStage = maxTasksPerStage;
+        this.nodeTaskMap = requireNonNull(nodeTaskMap, "nodeTaskMap is null");
+        this.queryManager = requireNonNull(queryManager, "queryManager is null");
+    }
+
+    @Override
+    public void lockDownNodes()
+    {
+        nodeMap.set(Suppliers.ofInstance(nodeMap.get().get()));
+        simpleNodeSelector.lockDownNodes();
+    }
+
+    @Override
+    public List<InternalNode> getActiveNodes()
+    {
+        return simpleNodeSelector.getActiveNodes();
+    }
+
+    @Override
+    public List<InternalNode> getAllNodes()
+    {
+        return simpleNodeSelector.getAllNodes();
+    }
+
+    @Override
+    public InternalNode selectCurrentNode()
+    {
+        return simpleNodeSelector.selectCurrentNode();
+    }
+
+    @Override
+    public List<InternalNode> selectRandomNodes(int limit, Set<InternalNode> excludedNodes)
+    {
+        Map<InternalNode, NodeTTL> nodeTTLInfo = ttlFetcherManager.getAllTTLs();
+
+        Map<InternalNode, Optional<ConfidenceBasedTTLInfo>> ttlInfo = nodeTTLInfo
+                .entrySet()
+                .stream()
+                .collect(toImmutableMap(
+                        Map.Entry::getKey,
+                        e -> e.getValue().getTTLInfo()
+                                .stream()
+                                .min(Comparator.comparing(ConfidenceBasedTTLInfo::getExpiryInstant))));
+
+        NodeMap nodeMap = this.nodeMap.get().get();
+        List<InternalNode> nodes = nodeMap.getActiveNodes();
+
+        ImmutableList<InternalNode> eligibleNodes = filterNodesByTTL(nodes, excludedNodes, ttlInfo);
+        return selectNodes(limit, new ResettableRandomizedIterator<>(eligibleNodes));
+    }
+
+    @Override
+    public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTask> existingTasks)
+    {
+        Multimap<InternalNode, Split> assignment = HashMultimap.create();
+        NodeMap nodeMap = this.nodeMap.get().get();
+        NodeAssignmentStats assignmentStats = new NodeAssignmentStats(nodeTaskMap, nodeMap, existingTasks);
+
+        List<InternalNode> eligibleNodes = getEligibleNodes(maxTasksPerStage, nodeMap, existingTasks);
+        NodeSelection randomNodeSelection = new RandomNodeSelection(eligibleNodes, minCandidates);
+
+        Set<InternalNode> blockedExactNodes = new HashSet<>();
```

Comment:
Looks like we are not populating this set. Do we want populate it based based on chosenNodeInfo ?

Reply:
The logic was present in `SimpleNodeScheduler` to support HARD_AFFINITY. The `SimpleTTLNodeScheduler` doesn't support HARD_AFFINITY, so we don't need the `blockedExactNodes`. I'll remove it.